source/core/src/main/com/csse3200/game/inventory/items/lootbox/configs/BaseLootTable.java

```diff
@@ -0,0 +1,86 @@
+package com.csse3200.game.inventory.items.lootbox.configs;
+
+import com.csse3200.game.inventory.items.AbstractItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * BaseLootTable is an abstract class representing a basic loot table used to manage and generate loot items.
+ * It provides functionality to add items with specified weights, retrieve single or multiple random items,
+ * and calculate their probabilities based on the assigned weights.
+ */
+public abstract class  BaseLootTable {
+    // List to hold loot items and their weights
+    protected List<LootItemConfig> items;
+
+    /**
+     * Constructs a BaseLootTable and initializes the list of items.
+     * This constructor sets up the internal structure needed to manage loot items.
+     */
+    protected BaseLootTable() {
+        this.items = new ArrayList<>();
+    }
+
+    /**
+     * Adds a loot item to the loot table with the specified weight and constructor parameters.
+     * The weight determines the probability of the item being selected when generating loot.
+     *
+     * @param item           The class type of the loot item (must extend AbstractItem).
+     * @param weight         The weight of the loot item, influencing its selection probability.
+     * @param parameterTypes The types of parameters needed for the item's constructor.
+     * @param parameters     The parameters to be passed when creating a new instance of the item.
+     */
+    public void addItem(Class<? extends AbstractItem> item, double weight, Class<?>[] parameterTypes,
+                        Object[] parameters) {
+        this.items.add(new LootItemConfig(item, weight, parameterTypes, parameters));
+    }
+
+    /**
+     * Retrieves the list of loot items in this loot table.
+     *
+     * @return List of LootItemConfig objects representing the items and their weights in the loot table.
+     */
+    public List<LootItemConfig> getItems() {
+        return this.items;
+    }
+
+    /**
+     * Selects a random item from the loot table based on the weights of each item.
+     * Items with higher weights have a greater probability of being selected.
+     *
+     * @return A new instance of a randomly selected AbstractItem, or null if an error occurs during instantiation.
+     */
+    public AbstractItem getRandomItem() {
+        double totalWeight = items.stream().mapToDouble(LootItemConfig::getWeight).sum();
+        double randomValue = Math.random() * totalWeight;
+        double cumulativeWeight = 0.0;
```

Comment:
Probably useful to create a ProbabilityDistribution class in utils and have a static function which takes in a list of probabilities and does random sampling for both any discrete distribution and bernoulli and binomial distributions. This will be useful for the map team when they need to do random sampling and for us when we improve random generation during chunk loading.